app/browser/reducers/passwordManagerReducer.js

```diff
@@ -190,53 +125,15 @@ const clearPasswords = () => {
   autofill.clearLogins()
 }
 
-const migrate = (state) => {
-  const passwords = state.get('passwords')
-  if (passwords.size) {
-    masterKey = masterKey || getMasterKey()
-    if (!masterKey) {
-      console.log('Could not access master password; aborting')
-      return
-    }
-    passwords.forEach((password) => {
-      let decrypted = CryptoUtil.decryptVerify(password.get('encryptedPassword'),
-                                               password.get('authTag'),
-                                               masterKey,
-                                               password.get('iv'))
-      if (decrypted) {
-        let form = {}
-        form['origin'] = password.get('origin')
-        form['signon_realm'] = password.get('origin') + '/'
-        form['action'] = password.get('action')
-        form['username'] = password.get('username')
-        form['password'] = decrypted
-        autofill.addLogin(form)
-      }
-    })
-    state = state.set('legacyPasswords', state.get('passwords'))
-    state = state.set('passwords', new Immutable.List())
-  }
-  const allSiteSettings = state.get('siteSettings')
-  const blackedList = allSiteSettings.filter((setting) => setting.get('savePasswords') === false)
-  if (blackedList.size) {
-    blackedList.forEach((entry, index) => {
-      let form = {}
-      form['origin'] = index
-      form['signon_realm'] = index + '/'
-      form['blacklisted_by_user'] = true
-      autofill.addLogin(form)
-      appActions.deletePasswordSite(index)
-    })
-  }
-  return state
-}
-
 const passwordManagerReducer = (state, action, immutableAction) => {
   action = immutableAction || makeImmutable(action)
   switch (action.get('actionType')) {
     case appConstants.APP_SET_STATE:
       init()
-      state = migrate(state)
```

Comment:
I think we should also check if `legacyPasswords` exists or not. If so, we should delete it to prevent our session data from getting bloated.